common/src/main/java/com/dumbdogdiner/stickyapi/common/configuration/providers/YamlProvider.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2020-2021 DumbDogDiner <dumbdogdiner.com>. All rights reserved.
+ * Licensed under the MIT license, see LICENSE for more information...
+ */
+package com.dumbdogdiner.stickyapi.common.configuration.providers;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+import com.dumbdogdiner.stickyapi.common.configuration.FileConfiguration;
+
+import org.apache.commons.lang.Validate;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.Yaml;
+
+public class YamlProvider implements FileConfiguration {
+
+    private Map<String, Object> data;
+
+    private InputStream inputStream;
+
+    public YamlProvider(InputStream stream) {
+        Validate.notNull(stream, "InputStream cannot be null!");
+
+        this.inputStream = stream;
+        this.reload();        
+    }
+
+    public YamlProvider(File file) throws FileNotFoundException {
+        this(new FileInputStream(file));
+    }
+
+    public void reload() {
+        // Attempt to (re)load the stream
```

Comment:
Interesting formatting going on here - we gonna have spotless, right?
```suggestion
        // Attempt to (re)load the stream
        Yaml yaml = new Yaml();
        Map<String, Object> obj = yaml.load(this.inputStream);
        this.data = obj;
```